diff --git a/linux/AmdMemTweak.cpp b/linux/AmdMemTweak.cpp
index f28ff54..39b075c 100644
--- a/linux/AmdMemTweak.cpp
+++ b/linux/AmdMemTweak.cpp
@@ -66,6 +66,7 @@ bool IsAmdDisplayDevice(struct pci_dev* dev)
 bool IsRelevantDeviceID(struct pci_dev* dev)
 {
 	return
+		(dev->device_id == 0x66a1) || // Vega 20 [Radeon Instinct MI50 32GB]
 		(dev->device_id == 0x66af) || // Radeon VII
 		(dev->device_id == 0x687f) || // Vega 10 XL/XT [Radeon RX Vega 56/64]
 		(dev->device_id == 0x6867) || // Vega 10 XL [Radeon Pro Vega 56]
@@ -1512,6 +1513,10 @@ int main(int argc, const char* argv[])
 		snprintf(buffer, sizeof(buffer) - 1, "/sys/kernel/debug/dri/%d/amdgpu_regs", instance);
 		gpu->mmio = open(buffer, O_RDWR);
 		if (gpu->mmio == -1)
+		{
+			gpu->mmio = open(buffer, O_RDONLY);
+		}
+		if (gpu->mmio == -1)
 		{
 			fprintf(stderr, "Failed to open %s\n", buffer);
 			return 1;
