diff --git a/requirements/rocm.txt b/requirements/rocm.txt
index 1fe1d8b51..3ca0e0cdf 100644
--- a/requirements/rocm.txt
+++ b/requirements/rocm.txt
@@ -16,3 +16,4 @@ runai-model-streamer[s3,gcs]==0.15.0
 conch-triton-kernels==1.2.1
 timm>=1.0.17
 fastsafetensors>=0.2.0
+flash-attn==2.8.3
diff --git a/vllm/model_executor/models/gemma3.py b/vllm/model_executor/models/gemma3.py
index 73176eba9..17365f5e8 100644
--- a/vllm/model_executor/models/gemma3.py
+++ b/vllm/model_executor/models/gemma3.py
@@ -344,6 +344,9 @@ class Gemma3DecoderLayer(nn.Module):
         residual: torch.Tensor | None,
         **kwargs,
     ) -> tuple[torch.Tensor, torch.Tensor]:
+        # https://github.com/huggingface/transformers/pull/36832
+        if hidden_states.dtype == torch.float16:
+            hidden_states = hidden_states.clamp_(-65504, 65504)
         if residual is None:
             residual = hidden_states
             hidden_states = self.input_layernorm(hidden_states)
@@ -355,12 +358,16 @@ class Gemma3DecoderLayer(nn.Module):
             **kwargs,
         )
         hidden_states = self.post_attention_layernorm(hidden_states)
+        if hidden_states.dtype == torch.float16:
+            hidden_states = hidden_states.clamp_(-65504, 65504)
 
         hidden_states, residual = self.pre_feedforward_layernorm(
             hidden_states, residual
         )
         hidden_states = self.mlp(hidden_states)
         hidden_states = self.post_feedforward_layernorm(hidden_states)
+        if hidden_states.dtype == torch.float16:
+            hidden_states = hidden_states.clamp_(-65504, 65504)
         return hidden_states, residual
 
 
